include/logging/log_core.h

```diff
@@ -125,6 +148,26 @@ extern "C" {
 	  (0)								     \
 	)
 
+/**
+ * @brief Macro for optional injection of function name as first argument of
+ *	  formatted string. _LOG_Z_EVAL() macro is used to handle no arguments
+ *	  case.
+ *
+ *	  The purpose of this macro is to prefix string literal with format
+ *	  specifier for function name and inject function name as first
+ *	  argument. In order to handle string with no arguments _LOG_Z_EVAL is
+ *	  used.
+ */
+#if CONFIG_LOG_FUNCTION_NAME
+#define _LOG_STR(...) "%s(): " __LOG_ARG_1(__VA_ARGS__), __func__\
```

Comment:
i was thinking that () will be self explained but if you think that it's not worth spending 2 characters i will remove it.

Reply:
I would keep keep the parentheses.

Reply:
@pizi-nordic I don't have a particularly strong opinion either way, just the general desire to keep the messages as compact as possible without sacrificing readability.

Reply:
@jhedberg i've noticed that #10711 @jukkar added %s() in case CONFIG_LOG_FUNCTION_NAME. I'm also leaning towards keeping parentheses (self-explaining as the argument).

Reply:
>  I'm also leaning towards keeping parentheses (self-explaining as the argument).

IMHO, drop it and save bytes. Likewise, intend to be propose to drop xml-like syntax for `<err>` vs `ERR`. (On critical path comparing that to the old log output.)

Reply:
Seems no approach to suit them all. However, i'm more convinced to change <err> to ERR that removing brackets. There is also instance level filtering. In that case messages are prefixed with <module_name>.<instance_name>.<function_name> (e.g. `foo.my_foo.init`) with those 3 words together brackets improve readability otherwise function name may be confused with instance name. Unless we use something else than `.` as the separator in one of those cases. but it has to be something that is not used in naming (`~`, `#` ?).

Some examples
- `foo.my_foo#init` and  `foo#init`
- `foo~my_foo.init` and `foo.init`

No strong opinion but imo any of those examples is less readable than `foo.my_foo.init()`.

Reply:
> No strong opinion but imo any of those examples is less readable than `foo.my_foo.init()`

Using dots to separate the fields looks less crowded and easier to read than the other options.